sm_ipt/template/sm_ipt_log_tmpl.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef SM_IPT_LOG_H_
+#define SM_IPT_LOG_H_
+
+/*
+ * THIS IS A TEMPLATE FILE.
+ * It should be copied to a suitable location within the host environment into
+ * which Remote Procedure serialization is integrated, and the following macros
+ * should be provided with appropriate implementations.
+ */
+
+/**
+ * @defgroup sm_ipt_log Logging functionality for SM IPT
+ * @{
+ * @ingroup sm_ipt
+ *
+ * @brief Logging functionality for SM IPT
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Macro for logging a message with the severity level ERR.
+ *
+ * @param ... printf-style format string, optionally followed by arguments
+ *            to be formatted and inserted in the resulting string.
+ */
+#define SM_IPT_ERR(...)
+
+/**
+ * @brief Macro for logging a message with the severity level WRN.
+ *
+ * @param ... printf-style format string, optionally followed by arguments
+ *            to be formatted and inserted in the resulting string.
+ */
+#define SM_IPT_WRN(...)
+
+/**
+ * @brief Macro for logging a message with the severity level INF.
+ *
+ * @param ... printf-style format string, optionally followed by arguments
+ *            to be formatted and inserted in the resulting string.
+ */
+#define SM_IPT_INF(...)
+
+/**
+ * @brief Macro for logging a message with the severity level DBG.
+ *
+ * @param ... printf-style format string, optionally followed by arguments
+ *            to be formatted and inserted in the resulting string.
+ */
+#define SM_IPT_DBG(...)
```

Comment:
We are not using logs different than `ERR`. Maybe it will be better to add some more logs, e.g.:
* `DBG` with block number (or location) and size for each allocation, deallocation, send, receive, e.t.c.
* `DBG` when there is not enough memory and we are waiting on semaphore for more free memory.
* `DBG` on each handshake step.
* `INF` with one line of information that module has been initialized and maybe some memory sizes information.